### PR TITLE
Go-Report-Card Fixes

### PR DESCRIPTION
**Pull request by otaviof**
# Changes

General fixes for go-report-card, typos and minor refactoring. Fixing the following alerts:

```
$ goreportcard-cli -v
2021/07/03 09:25:37 disabling misspell on large repo...
Grade: A+ (99.9%)
Files: 2043
Issues: 10
gofmt: 100%
go_vet: 100%
gocyclo: 99%
	pkg/shp/reactor/pod_watcher.go
		Line 59: warning: cyclomatic complexity 18 of function (*PodWatcher).Start() is high (> 15) (gocyclo)
golint: 99%
	pkg/shp/params/params.go
		Line 11: warning: comment on exported type Params should be of the form "Params ..." (with optional leading article) (golint)
		Line 53: warning: exported method Params.ClientSet should have comment or be unexported (golint)
	pkg/shp/flags/flags.go
		Line 13: warning: exported const BuildrefNameFlag should have comment (or a comment on this block) or be unexported (golint)
	pkg/shp/resource/doc.go
		Line 1: warning: package comment should be of the form "Package resource ..." (golint)
	pkg/shp/cmd/build/doc.go
		Line 1: warning: package comment should be of the form "Package build ..." (golint)
	pkg/shp/util/crud.go
		Line 43: warning: exported function UpdateObject should have comment or be unexported (golint)
	pkg/shp/util/doc.go
		Line 1: warning: package comment should be of the form "Package util ..." (golint)
	pkg/shp/resource/resource.go
		Line 51: warning: exported method Resource.Update should have comment or be unexported (golint)
		Line 94: warning: exported method Resource.Watch should have comment or be unexported (golint)
	pkg/shp/cmd/buildrun/buildrun.go
		Line 12: warning: exported function Command should have comment or be unexported (golint)
	pkg/shp/cmd/buildrun/doc.go
		Line 1: warning: package comment should be of the form "Package buildrun ..." (golint)
license: 100%
ineffassign: 100%
misspell: 100%
```

# Submitter Checklist

- [x] ~Includes tests if functionality changed/was added~
- [x] ~Includes docs if changes are user-facing~
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [x] Release notes block has been filled in, or marked NONE

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/shp/reactor/pod_watcher.go
Comment (coreydaley) [inline pkg/shp/reactor/pod_watcher.go:103]: This looks more like a refactor than part of the Go report card fixes. Was this flagged by the tool?

Comment (otaviof) [reply pkg/shp/reactor/pod_watcher.go:103]: Indeed a minor refactoring due to high cyclomatic-complexity, i.e.:

```
$ goreportcard-cli -v
[...]
	pkg/shp/reactor/pod_watcher.go
		Line 59: warning: cyclomatic complexity 18 of function (*PodWatcher).Start() is high (> 15) (gocyclo)
[...]
```

**Comment by gabemontero** [comment]
generally looks good to me, but I'll defer to @coreydaley / @alicerum on completing the reivew process and tagging this

**Comment by coreydaley** [comment]
Looks like I'm not on the reviewers or the approvers list @otaviof https://github.com/shipwright-io/cli/blob/main/OWNERS
Someone else will have to do the approval

**Comment by gabemontero** [comment]
> Looks like I'm not on the reviewers or the approvers list @otaviof https://github.com/shipwright-io/cli/blob/main/OWNERS
> Someone else will have to do the approval

ok we can get into updating the owners file once you start hacking on this repo @coreydaley ... maybe as part of whatever initial PR you submit

that said, with @otaviof being listed as an approve, you should be able to "self approve" ... approve works differently that lgtm ... but I'm pretty sure that Adam set things up so that approvers' PRs are not automatically approved (vs. how openshift repos work).

Give specifying the approve prow command a try @otaviof and we'll go from there

**Review by alicerum — APPROVED** [review]

**Pull request merged**
